### PR TITLE
[velux] fix concurrency bugs & other minor issues; and update readme.md file

### PR DESCRIPTION
**Pull request by andrewfg**
The binding would not support multiple KLF bridges concurrently.
This PR fixes that problem and a few other minor bugs.
Also updated the ReadMe.md file.
Work-in-Progress => needs review by (author) Guenther Schreiner!

### CODE REVIEW

**Comment by andrewfg** [comment]
@gs4711 just to let you know that I created this PR :)

File: bundles/org.openhab.binding.velux/src/main/java/org/openhab/binding/velux/internal/bridge/json/JsonBridgeAPI.java
Comment (Hilbrand) [inline bundles/org.openhab.binding.velux/src/main/java/org/openhab/binding/velux/internal/bridge/json/JsonBridgeAPI.java:86]: You should probably change the casing If these variables are not static final anymore. They should not be uppercase.

Comment (andrewfg) [reply bundles/org.openhab.binding.velux/src/main/java/org/openhab/binding/velux/internal/bridge/json/JsonBridgeAPI.java:86]: Good point. Although I would like to hear Guenther’s comments on the fundamentals of the change from using ‘static’ before I address the code styling.

**Review by gs4711 — APPROVED** [review]

**Review by gs4711 — COMMENTED** [review]
[velux] flush read buffer before sending command 

This fix could lead to some problems if there are pending unprocessed messages delivered to the binding which are expected in long running methods i.e. the actuar manipulation. 

For now, I'd suggest to keep it but to make it visible to the user (i.e. by logging) that there is a discard of answers.

**Comment by gs4711** [comment]
@andrewfg ,
I'm sorry replying too late, but I messed up my development environment when I tried to work on version 2.5.x and 3.0 in parallel. It will still take some time to start over.

File: bundles/org.openhab.binding.velux/src/main/java/org/openhab/binding/velux/internal/bridge/json/JsonBridgeAPI.java
Comment (andrewfg) [reply bundles/org.openhab.binding.velux/src/main/java/org/openhab/binding/velux/internal/bridge/json/JsonBridgeAPI.java]: Ok. I changed the capitalisation.

**Comment by andrewfg** [comment]
> [velux] flush read buffer before sending command
> 
> This fix could lead to some problems if there are pending unprocessed messages delivered to the binding which are expected in long running methods i.e. the actuar manipulation.
> 
> For now, I'd suggest to keep it but to make it visible to the user (i.e. by logging) that there is a discard of answers.

Noted. However I think that the way you have serialized your threads means that under normal operation -- even on long running methods -- the next thread will not be started until the `_FINISHED_NTF` has been received on the prior thread.

**Comment by andrewfg** [comment]
**Warning**: although I tested my code thoroughly yesterday, I left it running overnight to be sure, and unfortunately found some new error messages in the log this morning. 

**=> PLEASE DO NOT MERGE (YET)**

**Comment by andrewfg** [comment]
> kaikreuzer force-pushed the openhab:2.5.x branch from 3afd2bd to 4cb13dc 13 hours ago

@kaikreuzer I set the `migrate to OH3` tag after you closed this PR, in the hope of starting the OH3 migration workflow; however it seems that tag only starts the workflow on open PRs. => so can you kindly please reopen this PR to let the migration workflow run? Or otherwise run the migration workflow for me?

**Comment by kaikreuzer** [comment]
I cannot reopen it either. You might want to try to rebase the branch as it at the moment claims that there is no history in common and thus cannot open the PR.

**Comment by andrewfg** [comment]
@kaikreuzer 

> try to rebase the branch as it at the moment claims that there is no history in common and thus cannot open the PR.

I tried that with the following commands, but then it asks me to manually resolve **5028** conflicts. (And for sure, I am not willing to do that...)

```
git checkout velux
git pull --rebase upstream main
```
I suppose that I am not the only addons developer to have been ambushed by this problem, so a fix from you guys (core team) would be very much appreciated.

**Comment by Hilbrand** [comment]
I can fix this for you. But at the moment the main branch is still not completely up-and-running so it's best to wait for this to finish before fixing it here to avoid unnecessary conflicts. I think we'll also write up some more documentation on how to proceed from the original situation to the new, I'm just gathering that information.

**Comment by andrewfg** [comment]
@kaikreuzer just a guess, but I suppose it may be related to the fact that my repo is a branch from a branch from a fork on the `master` branch of openHAB ( `openhab:master=>2.5.x=>andrewfg:master=>2.5.x=>velux` ) and on your side you changed from `master ` to `main` so gitHub has "munged" the linkage between work on my repo and the "new" OH3  :(

**Comment by andrewfg** [comment]
Recreated this PR in #8520

**Pull request closed without merging**
